src/tool/subcommands/benchmark_cmd.rs

```diff
@@ -135,17 +149,79 @@ async fn benchmark_car_streaming_inspect(input: Vec<PathBuf>) -> Result<()> {
             .and_then(CarStream::new)
             .try_flatten(),
     );
+    let mut seen = CidHashSet::default();
     while let Some(block) = s.try_next().await? {
         let block: Block = block;
         if block.cid.codec() == DAG_CBOR {
             let cid_vec: Vec<Cid> = extract_cids(&block.data)?;
-            let _ = cid_vec.iter().unique().count();
+            for cid in cid_vec {
+                if !should_save_block_to_snapshot(cid) {
+                    continue;
+                }
+                seen.insert(cid);
+            }
         }
         sink.write_all(&block.data).await?
     }
     Ok(())
 }
 
+// Concatenate a set of CAR files and measure how quickly we can stream the
+// blocks, while inspecting them in parallel.
+// NOTE: when testing with `cargo forest-tool benchmark unordered-graph-traversal forest_snapshot_mainnet_2023-05-31_height_2908403.car`
+// there is a performance dip starting around 60+GB processed, that later fixes itself. I can see
+// this pattern for normal car streaming too of course. Would be nice to understand what's going on.
+async fn benchmark_parallel_car_streaming_inspect(input: Vec<PathBuf>) -> Result<()> {
+    let mut sink = indicatif_sink("traversed");
+    let mut s = Box::pin(
+        futures::stream::iter(input)
+            .then(File::open)
+            .map_ok(BufReader::new)
+            .and_then(CarStream::new)
+            .try_flatten(),
+    );
+
+    let (limit_sender, limit_receiver) = flume::bounded(num_cpus::get() * 5);
+    let (sender, receiver) = flume::bounded(4096);
+
+    let seen = Arc::new(Mutex::new(CidHashSet::default()));
```

Comment:
This is a draft, excuses.
We don't need the `RwLock` here, because there's only one thread. The `Mutex` is only there so that we can later print out `seen` from the main thread, the isn't any concurrency going on. 
For this scenario `Mutex` has the optimal performance, I haven't measured `RwLock`, but I assume it should probably be worse or similar.

Reply:
Some curious reading on the subject: https://fy.blackhats.net.au/blog/2018-10-19-rust-rwlock-and-mutex-performance-oddities/#:~:text=In%20the%20process%20I%20wanted,the%20same%20%2D%202%20times%20faster.